File: easy/121/max_profit.ts
function maxProfit(prices: number[]): number {
  let profit: number = 0;

  for (let buy: number = 0; buy < prices.length; buy++) {
    for (let sell: number = buy + 1; sell < prices.length; sell++) {
      let tmp_profit: number = prices[sell] - prices[buy];
      if (tmp_profit <= 0) {
        break;
      }
      if (tmp_profit > profit) {
        profit = tmp_profit;
      }
    }
  }

  return profit;
}

File: easy/121/max_profit.go
package main

func maxProfit(prices []int) int {
	if len(prices) == 1 {
		return 0
	}

	min_price := prices[0]
	profit := 0
	for i := 1; i < len(prices); i++ {
		if min_price >= prices[i] {
			min_price = prices[i]
		} else if (prices[i] - min_price) > profit {
			profit = prices[i] - min_price
		}
	}

	return profit
}
